ocaml/xapi/xapi_pool.ml

```diff
@@ -3515,12 +3515,25 @@ let remove_repository ~__context ~self ~value =
   if Db.Pool.get_repositories ~__context ~self = [] then
     Db.Pool.set_last_update_sync ~__context ~self ~value:Date.epoch
 
-let sync_repos ~__context ~self ~repos ~force ~token ~token_id =
+let sync_repos ~__context ~self ~repos ~force ~token ~token_id ~username
+    ~password =
+  let remote_addr =
+    let repo =
+      Repository_helpers.get_single_enabled_update_repository ~__context
+    in
+    match Repository_helpers.is_remote_pool_repository ~__context ~repo with
+    | true ->
+        Db.Repository.get_binary_url ~__context ~self:repo
+        |> Repository_helpers.get_remote_pool_coordinator_ip
+    | false ->
```

Comment:
Here the `remote_addr` could be an empty string, which will be passed onto `sync -> ext_host_verified_rpc -> make_external_host_verified_rpc -> make_remote_rpc -> XML_protocol.rpc` what is the consequence of an empty `remote_addr` downstream?